File: fizzbuzz/java/fizzbuzz.java

class FizzBuzz {

    public static void main(String args[]) {

        if (args.length < 1) {
            System.out.println(Integer.toString(args.length));
            System.out.println("Usage: ./fizzbuzz <num>");
        }
        else {
            int high = Integer.parseInt(args[0]);
            for (int ii = 1; ii <= high; ii++) {
                String aa = "";
                if (ii % 3 == 0) {
                    aa += "fizz";
                }
                if (ii % 5 == 0) {
                    aa += "buzz";
                }

                if (!aa.equals("")) {
                    System.out.println(aa);
                } else {
                    System.out.println(Integer.toString(ii));
                }
            }
        }

    }

}

File: fizzbuzz/rust/fizzbuzz.rs
use std::env;

fn main() {
    let args: Vec<String> = env::args().collect();
    let top = args.get(1).expect("Usage ./fizzbuzz <num>");
    let top: u32 = top.parse().expect("Not a number!");

    for ii in 1..top {
        let mut acc : String = "".to_string();
        if ii % 3 == 0 {
            acc.push_str("fizz");
        }

        if ii % 5 == 0 {
            acc.push_str("buzz");
        }

        if acc == "" {
            println!("{}", ii);
        }
        else {
            println!("{}", acc);
        }
    }
    println!("{}", top);
}
